### DISCUSSION TITLE
Swift Package Manager Wrapping Google Maps XCFramework

### DISCUSSION THREAD

**Discussion by Adekunle77** [firebase/firebase-ios-sdk#10243] [Q&A]
I have downloaded the Google Maps XCFramework from [here](https://developers.google.com/maps/documentation/ios-sdk/config#xcframework-manual) 

And I created a fresh Swift Package Manage project and drop the XCFrameworks into the project;t GoogleMapsBase.xcframework,GoogleMaps.xcframework, andGoogleMapsCore.xcframework

My Package.swift looks like this 
![Screenshot 2022-09-21 at 18 52 01-1](https://user-images.githubusercontent.com/14952997/191608019-a69002ef-05ba-4149-92c6-8e9814a7897e.jpg)

When I drag the completed SPM with the XCFrameworks in an Xcode project and add the XCFrameworks in the Frameworks, Libraries and Embedded Content.

When I run the project it complies but it does not build or install on the simulator and I get this message 
![191607915-bf2db4c4-411d-4108-8529-45decbdd3ea9](https://user-images.githubusercontent.com/14952997/192702577-cb077e4b-cc9f-4f6b-8ed9-7ce4070fd9b7.jpg)

I do not know how to solve it. Or is there another way to wrap the Google Maps XCFramework using Swift Package Manager?

Thanks
Ade

**Comment by paulb777** [comment]
Here's another discussion and proposed solution https://forums.swift.org/t/swift-package-for-google-maps-ios-sdk/42325

**Reply by Adekunle77** [reply]
@paulb777 I have to use this but it needs to be updated. Anyways would like to know how I can develop my own. @paulb777 Can I get your email so I can send you the whole Swift Package project so you can see where I am going wrong, double please.  here it is on git. https://github.com/Adekunle77/SPMGoogleMaps/tree/master

**Reply by paulb777** [reply]
@Adekunle77 Sorry, but I don't have the bandwidth to do support on non-Firebase products. I recommend you review [Apple's documentation](https://developer.apple.com/documentation/xcode/distributing-binary-frameworks-as-swift-packages)  and ask a question on stackoverflow.

Also, I've only done the checksum version of SPM binary resources like [here](https://github.com/firebase/firebase-ios-sdk/blob/master/Package.swift#L327).
